### ISSUE TITLE
Undefined method `createdTimestamp()` when called on author

### ISSUE THREAD

**Issue by RBotfield** [discord-php/DiscordPHP#628]
**Environment**
- PHP Version:
    - 8.0.13
    - 8.1.0RC6 
- DiscordPHP Version:
    - dev-master (ref: 5f3717a11796d97ab263300d4655efed2c122802)

**Describe the bug**
Calling `createdTimestamp()` on the `author` property of a `Message` part results in an undefined method error since it attempts to pass that function down to the `Member` part where it doesn't exist as a method.
It _does_ however exist for the `User` part, so calling as `$message->user->createdTimestamp()` is a valid workaround for now.

```
Undefined method Discord\Parts\User\Member::createdTimestamp()
```

**To Reproduce**
```php
$discord->on(Evene::MESSAGE_CREATE, function (Message $message, Discord $client) {
    echo $message->author->createdTimestamp();
});
```

**Expected behavior**
`createdTimestamp()` should behave the same way on `Member` parts as it should on `User`, or the function should be _only_ called against the user.

**Screenshots**

**Additional context**

**Comment by SQKo** [comment]
> It _does_ however exist for the `User` part, so calling as `$message->user->createdTimestamp()` is a valid workaround for now.

That's not the workaround, that's what you supposed to use.

Even with Member object you will have to go with `$member->user->createdTimestamp()`

**Comment by davidcole1340** [comment]
Unfortunately it doesn't really make sense for `Member` to have the same `createdTimestamp()` function as `User`, as a member is created when you join a guild, while a user is created when you join Discord.

Problems like these have always been around with DiscordPHP as `author` can return `User` or `Member`, so moving forwards we recommend dropping the usage of the `author` attribute and using the `user` or `member` attributes. The `author` attribute is there for backwards compatibility at this point.

**Comment by RBotfield** [comment]
> That's not the workaround, that's what you supposed to use.

Considering the parts for this project model Discord's returned objects pretty well, I had expected to use `$message->author` over `$message->user` (source: discord's current docs for Message objects)
Even still, noted & I'll avoid using the author property where I can especially as I know that's a different issue entirely

However in this specific case, which currently typehints as a valid call until you find out in runtime, should the call-chain instead of being:
`$message->author->member->createdTimestamp()` 
could instead be changed to
`$message->author->member->user->createdTimestamp()`

Could be implemented by adding a method to the Member class
```
public function createdTimestamp()
{
    $this->user->createdTimestamp();
}
```

**Comment by davidcole1340** [comment]
The original thought with `author` vs `user` is to deprecate `author` instead of making a breaking change, however, since the next version is going to be a breaking release anyway I'm pushing further to make `author` return a `User`, while `member` will return a `Member`.

We could add a forwarding method like shown, but one would think that `$member->createdTimestamp()` would think it would return the creation date of the member, which is when the user joined the guild, while it would actually return the creation of the `User` (at least as someone that understands the Discord data models).

**Comment by SQKo** [comment]
This should have been resolved with #663 because `$message->author` now only returns `User` object

As David pointed out, we will not add `createdTimeStamp()` on `Member`, you should either use `$user->createdTimeStamp()` for the User discord joined time, and `$member->joined_at` for the Member guild joined time.
